### ISSUE TITLE
Transpile bug when using sx basis with Y and -Y

### ISSUE THREAD

**Issue by naezzell** [Qiskit/qiskit-terra#5742]
### Informations

- **Qiskit version**: 0.23.2
- **Python version**: 3.9.1
- **Operating system**: 10.15.7

### What is the current behavior?
Transpile gives wrong native circuit with sx basis but correct circuit with u basis for Armonk.

### Steps to reproduce the problem
We will define a 180 degree rotation about y axis (i.e. the Y gate) and about the -y axis (i.e. -Y gate) using the decomposition of an arbitrary rotation.

```
# First the Y circuit
y_circ = QuantumCircuit(1, 1)
y_circ.barrier(0)
y_circ.rz(-np.pi / 2, 0)
y_circ.ry(-np.pi / 2, 0)
y_circ.rz(np.pi, 0)
y_circ.ry(np.pi / 2, 0)
y_circ.rz(np.pi / 2, 0)
y_circ.barrier(0)

# Now the -Y circuit
my_circ = QuantumCircuit(1, 1)
my_circ.barrier(0)
my_circ.rz(-3*np.pi / 2, 0)
my_circ.ry(-np.pi / 2, 0)
my_circ.rz(np.pi, 0)
my_circ.ry(np.pi / 2, 0)
my_circ.rz(3*np.pi / 2, 0)
my_circ.barrier(0)
```

Now we will transpile these using Armonk as the backend (1) with the sx basis and (2) with the u basis.
```
y_tcirc = transpile(y_circ, armonk)
y_tcirc.draw()
```
<img width="566" alt="Screen Shot 2021-01-28 at 12 07 16 PM" src="https://user-images.githubusercontent.com/29308150/106192684-61796600-6161-11eb-8630-05e7e9399523.png">

```
my_tcirc = transpile(my_circ, armonk)
my_tcirc.draw()
```
<img width="547" alt="Screen Shot 2021-01-28 at 12 10 43 PM" src="https://user-images.githubusercontent.com/29308150/106193131-f3816e80-6161-11eb-8cd2-a2524157ad32.png">

```
y_tcirc_u = transpile(y_circ, armonk,  basis_gates=['u1', 'u2', 'u3', 'cx'])
y_tcirc_u.draw()
```
<img width="528" alt="Screen Shot 2021-01-28 at 12 07 57 PM" src="https://user-images.githubusercontent.com/29308150/106192936-b1582d00-6161-11eb-9845-fdef6d275f28.png">

```
my_tcirc_u = transpile(my_circ, armonk, basis_gates=['u1', 'u2', 'u3', 'cx'])
my_tcirc_u.draw()
```
<img width="528" alt="Screen Shot 2021-01-28 at 12 10 24 PM" src="https://user-images.githubusercontent.com/29308150/106193176-03994e00-6162-11eb-9af8-478912149fe7.png">

Notice that the sx basis gives same exact transpiled circuit whereas u basis differs in the argument of u3. If you simulate the circuit, they differ by a global phase (i.e. whether you get i|1> or -i|1>), and the phase difference between Y and -Y is important, i.e. for DD, so this isn't good.

Note: You can do the same procedure with X and -X, but you actually get a difference in both bases as expected.

### What is the expected behavior?
Using the sx basis shouldn't destroy phase information that the u basis preserves, as this is important for implementing real procedures. 

### Suggested solutions
I'm not sure how transpile works under the hood, so it's hard to offer suggested solution. However, if it's expected that sx basis may loose phase information, it would be nice for a warning to be given. 

P.S. If I specify things in u basis, are they still transpiled to sx before being run, and if so, how can I implement Y and -Y as I want?

**Comment by mtreinish** [comment]
@naezzell have you tried this on qiskit 0.23.3 or 0.23.4 (which was just released a little while ago)? 0.23.3 included some fixes for the transpiler around global phase tracking (see https://github.com/Qiskit/qiskit-terra/pull/5618 and https://github.com/Qiskit/qiskit-terra/pull/5474). I haven't looked at this issue in detail yet, so I'm not sure if it will fix it, or there is something else going on.

**Comment by naezzell** [comment]
> @naezzell have you tried this on qiskit 0.23.3 or 0.23.4 (which was just released a little while ago)? 0.23.3 included some fixes for the transpiler around global phase tracking (see #5618 and #5474). I haven't looked at this issue in detail yet, so I'm not sure if it will fix it, or there is something else going on.

Things seem to work in qiskit 0.23.4. Thanks for the heads up!
